app/scripts/lib/stores/read-only-network-store.test.ts

```diff
@@ -0,0 +1,141 @@
+import log from 'loglevel';
+import nock from 'nock';
+import ReadOnlyNetworkStore from './read-only-network-store';
+import { MetaMaskStorageStructure } from './base-store';
+
+const FIXTURE_SERVER_HOST = 'localhost';
+const FIXTURE_SERVER_PORT = 12345;
+const FIXTURE_SERVER_ORIGIN = `http://${FIXTURE_SERVER_HOST}:${FIXTURE_SERVER_PORT}`;
+const FIXTURE_SERVER_PATH = '/state.json';
+
+const DEFAULT_INITIAL_STATE = {
+  data: { config: {} },
+};
+
+const MOCK_STATE = { data: { config: { foo: 'bar' } }, meta: { version: 1 } };
+
+/**
+ * Initiatilizes a ReadOnlyNetworkStore for testing
+ *
+ * @returns store - a ReadOnlyNetworkStore
+ */
+function setupReadOnlyNetworkStore() {
+  const store = new ReadOnlyNetworkStore();
+  return store;
+}
+
+/**
+ * Create a Nock scope for the fixture server response.
+ *
+ * @returns A Nock interceptor for the fixture server response.
+ */
+function mockFixtureServerInterceptor(): nock.Interceptor {
+  return nock(FIXTURE_SERVER_ORIGIN).get(FIXTURE_SERVER_PATH);
+}
+
+/**
+ * Create a Nock scope for the fixture server response, which will have a successful reply.
+ *
+ * @param state
+ */
+function setMockFixtureServerReply(
+  state: Record<string, unknown> = DEFAULT_INITIAL_STATE,
+): void {
+  mockFixtureServerInterceptor().reply(200, state);
+}
+
+describe('ReadOnlyNetworkStore', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    nock.cleanAll();
+  });
+
+  describe('constructor', () => {
+    it('makes a request to the fixture server upon construction', async () => {
+      const scope = nock(FIXTURE_SERVER_ORIGIN)
+        .get(FIXTURE_SERVER_PATH)
+        .reply(200, 'Test');
+      expect(scope.pendingMocks().length).toBe(1);
+      setupReadOnlyNetworkStore();
+      await new Promise(process.nextTick);
+      expect(scope.pendingMocks().length).toBe(0);
+    });
+  });
+
+  describe('initialization (#init)', () => {
```

Comment:
Nit: May as well leave these in the same block, since the code-under-test is the constructor for both.

No need to embed internal implementation details in the tests, that just makes them more likely to have to change if we refactor things.

```suggestion

```